### PR TITLE
Added routeName parameter to WrapperFn

### PR DESCRIPTION
**Pull request by diefferson**
Hello, this library saved my life.
Thank you so much for sharing it.

But there is a scenario in my project where I adapted it and I think it might be interesting to be added to it.

In a Router we can define a ScreensWrapper that envelops all the screens of that router in a standard widget which is very cool.
However I needed to know the what the route at that moment  will encapsulate each screen.

For example, in my project I use the lib [Stark](https://pub.dev/packages/stark) to do the injection of dependencies and in it I can create a Scope for my widget so that the injections with that scope are poured out automatically:
```dart
  @override
  WrapperFn get screensWrapper =>
      (BuildContext context, Widget child, String routeName) {
        return Scope(
          name: routeName,
          child: child,
        );
      };
```

This also allows for countless other things like having unique Keys for each screen.

### CODE REVIEW

File: lib/src/nuvigator.dart [hunk 1]
```diff
@@ -18,6 +18,7 @@ NuvigatorState _tryToFindNuvigatorForRouter<T extends Router>(
 
 class NuvigatorStateTracker extends NavigatorObserver {
   final List<Route> stack = [];
+  String currentRouteName;
```
Comment (leoiacovini) [inline lib/src/nuvigator.dart:21]: I would avoid adding a new field for this when it could just be a getter

File: lib/src/nuvigator.dart [hunk 2]
```diff
@@ -346,7 +351,7 @@ class NuvigatorState<T extends Router> extends NavigatorState
   Widget build(BuildContext context) {
     Widget child = super.build(context);
     if (widget.wrapper != null) {
-      child = widget.wrapper(context, child);
+      child = widget.wrapper(context, child, stateTracker.currentRouteName);
```
Comment (leoiacovini) [inline lib/src/nuvigator.dart:354]: Passing the routeName here does not make much sense given this wrapper is applied around the Navigator, and not on a specific screen

File: lib/src/screen_route.dart [hunk 3]
```diff
@@ -4,7 +4,8 @@ import 'package:nuvigator/nuvigator.dart';
 
 import 'screen_type.dart';
 
-typedef WrapperFn = Widget Function(BuildContext context, Widget child);
+typedef WrapperFn = Widget Function(
+    BuildContext context, Widget child, String routeName);
```
Comment (leoiacovini) [inline lib/src/screen_route.dart:8]: This is a breaking change that would be very good if we could avoid

**Comment by diefferson** [comment]
@leoiacovini Yes, I can use ModalRoute in my case, thanks.
I updated the code for this pull request only with wrapper tests that had not yet been done. if you want add this tests to library.

**Pull request merged**